module/zfs/dmu.c

```diff
@@ -153,6 +161,34 @@ const dmu_object_byteswap_info_t dmu_ot_byteswap[DMU_BSWAP_NUMFUNCS] = {
 	{	zfs_acl_byteswap,	"acl"		}
 };
 
+/*
+ * Checking to see if a Direct IO write operation has updated the
+ * db_blkptr. We must pull the updated block pointer to get the
+ * current version of the block if a Direct IO write has occurred.
```

Comment:
This seems a little vague, how about something like this:

* Normally the db_blkptr points to the most recent on-disk content for the dbuf (and anything
* newer will be cached in the dbuf). However, a recent Direct IO write could leave newer content
* on disk and the dbuf uncached. In this case we must return the (as yet unsynced) pointer to the
* newer on-disk content.

Reply:
The comment has been updated in dmu_direct.c.